### DISCUSSION TITLE
How to get FPS for a model?

### DISCUSSION THREAD

**Discussion by kirkzZ** [facebookresearch/detectron2#3837] [Q&A]
How to get FPS for a model?

## Note

We only consider adding new features if they are relevant to many users.

If you request implementation of research papers -- we only consider papers that have enough significance and prevalance in the object detection field.

We do not take requests for most projects in the `projects/` directory, because they are research code release that is mainly for other researchers to reproduce results.

"Make X faster/accurate" is not a valid feature request. "Implement a concrete feature that can make X faster/accurate" can be a valid feature request.

Instead of adding features inside detectron2,
you can implement many features by [extending detectron2](https://detectron2.readthedocs.io/tutorials/extend.html).
The [projects/](https://github.com/facebookresearch/detectron2/tree/main/projects/) directory contains many of such examples.

**Comment by zhanghang1989** [comment]
For fps on GPU, it is shown in the training log. We don't have open source version of benchmarking on devices.

**Comment by kirkzZ** [comment]
hello, thanks for your answering, the time is the fps?                                                                                                                     
eta: 9:58:17  iter: 174159  total_loss: 0.9185  loss_cls: 0.229  loss_box_reg: 0.2562  loss_mask: 0.2784  loss_rpn_cls: 0.04507  loss_rpn_loc: 0.0934  time: 0.3831  data_time: 0.0075  lr: 0.01  max_mem: 3067M

**Comment by zhanghang1989** [comment]
you may infer the fps based on `fps = batch_size / time`

**Comment by kirkzZ** [comment]
If I set the SOLVER.IMS_PER_BATCH', '8'  ,so the fps = 8/time

**Comment by zhanghang1989** [comment]
You may also need to divide it by number of gpus, if more than one is using.

**Comment by kirkzZ** [comment]
Thanks,in a word:
 if I set  num_gpus=4 and SOLVER.IMS_PER_BATCH', '8',
and get the log:eta: 9:58:17 iter: 174159 total_loss: 0.9185 loss_cls: 0.229 loss_box_reg: 0.2562 loss_mask: 0.2784 loss_rpn_cls: 0.04507 loss_rpn_loc: 0.0934 time: 0.3831 data_time: 0.0075 lr: 0.01 max_mem: 3067M

so the fps=8/(4*0.3831)
